Mathlib/Algebra/Ring/UnitEquivs.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez, Johan Commelin
+-/
+import Mathlib.GroupTheory.GroupAction.ConjAct
+import Mathlib.RingTheory.Subring.Basic
+
+/-!
+# Equivalences relating to units
+
+This file makes a couple equivalences that deal with unit groups under various other operations,
+such as centralizers and stabilizers. These are mainly useful to prove Wedderburn's little theorem.
+
+-/
+
+variable {D : Type*}
+
+/-- The stabilizer of `Dˣ` acting on itself by conjugation at `x : Dˣ` is exactly the
+units of the centralizer of `x : D`. -/
+def unitsCentralizerEquiv [Monoid D] (x : Dˣ) :
+    MulAction.stabilizer (ConjAct Dˣ) x ≃* (Submonoid.centralizer ({↑x} : Set D))ˣ where
+  toFun := MonoidHom.toHomUnits <|
+  ⟨⟨fun u ↦ ⟨↑(ConjAct.ofConjAct u.1 : Dˣ), by
+    rintro x ⟨rfl⟩; have : _ • _ = _ := u.2
+    rwa [ConjAct.smul_def, mul_inv_eq_iff_eq_mul, Units.ext_iff, eq_comm] at this⟩, rfl⟩,
+    fun a b ↦ rfl⟩
+  invFun u := ⟨ConjAct.toConjAct (Units.map (Submonoid.centralizer ({↑x} : Set D)).subtype u), by
+    change _ • _ = _
+    simp only [ConjAct.smul_def, ConjAct.ofConjAct_toConjAct, mul_inv_eq_iff_eq_mul]
+    exact Units.ext <| (u.1.2 x <| Set.mem_singleton _).symm⟩
+  left_inv _ := by ext; rfl
+  right_inv _ := by ext; rfl
+  map_mul' := map_mul _
+
+/-- `unitsCentralizerEquiv` for a `Ring`. Restated for Wedderburn's little theorem. -/
+def unitsCentralizerEquivSubring [Ring D] (x : Dˣ) :
+    MulAction.stabilizer (ConjAct Dˣ) x ≃* (Subring.centralizer ({↑x} : Set D))ˣ :=
+  unitsCentralizerEquiv x
+
```

Comment:
I don't understand why we need this. Can you not just use what you have for `Monoid`s?